### ISSUE TITLE
Support gradient checkpointing for ESM models

### ISSUE THREAD

**Issue by mahdip72** [huggingface/transformers#24602]
Would you please add `gradient_checkpointing_enable()` feature for ESM models?
These models currently are the best available pre-trained protein language models for researchers.
Many thanks.

**Comment by amyeroberts** [comment]
cc @Rocketknight1

**Comment by mahdip72** [comment]
Any updates?

**Comment by Rocketknight1** [comment]
It's on the to-do list, but I'm afraid there are competing priorities at the moment!

**Comment by amyeroberts** [comment]
Let's open it up for anyone in the community who might want to tackle it :)

**Comment by sanjeevk-os** [comment]
Hi @amyeroberts @Rocketknight1 I would like to work on this

**Comment by Rocketknight1** [comment]
Hi @sanjeevk-os, I actually took a look at the ESM code - it actually looks like some of the supports for gradient checkpointing are already there, in which case you just need to make a one-line change to set `supports_gradient_checkpointing = True`

**Comment by sanjeevk-os** [comment]
Hi @Rocketknight1 Thank you for taking a look. I also noticed that the ESM model has the _create_custom_forward_ passed to torch checkpoint function. I will do some more checks and will raise a PR soon.

**Comment by Rocketknight1** [comment]
Hi @sanjeevk-os - we're getting even more requests for this, so we'd like to try to add it soon! If you're having trouble, just let us know. We can take over the PR internally to try to get it through, and we appreciate your effort regardless.

**Comment by Rocketknight1** [comment]
This issue has now been resolved - thank you to @sanjeevk-os for the very clean PR!
